src/signals-win.c

```diff
@@ -326,18 +326,47 @@ volatile HANDLE hBtThread = 0;
 static DWORD WINAPI profile_bt( LPVOID lparam )
 {
     // Note: illegal to use jl_* functions from this thread
-
     TIMECAPS tc;
     if (MMSYSERR_NOERROR!=timeGetDevCaps(&tc, sizeof(tc))) {
         fputs("failed to get timer resolution",stderr);
         hBtThread = 0;
         return 0;
     }
+    //Precision timer
+    LARGE_INTEGER StartingTime, EndingTime, ElapsedMicroseconds;
+    LARGE_INTEGER Frequency;
+    QueryPerformanceFrequency(&Frequency);
     while (1) {
         if (running && bt_size_cur < bt_size_max) {
-            DWORD timeout = nsecprof/GIGA;
-            timeout = min(max(timeout,tc.wPeriodMin*2),tc.wPeriodMax/2);
-            Sleep(timeout);
+            //Convert timeout to microseconds
+            DWORD timeout = nsecprof/1000;
+            //if timeout is greater than a millisecond use Sleep
+            if (timeout < 1000)
+            {
+                //calculate start time for high precision sleep
+                QueryPerformanceCounter(&StartingTime);
+                int time_elapsed = 0;
+                while (time_elapsed ==0)
```

Comment:
very minor, but would look a little nicer with spaces on both sides of the `==`, not just on the left